pkg/workloads/spark_job/spark_util.py

```diff
@@ -220,53 +229,61 @@ def ingest(ctx, spark):
         df = read_parquet(ctx, spark)
 
     input_type_map = {f.name: f.dataType for f in df.schema}
-
-    for raw_column_name in ctx.raw_columns.keys():
+    inferred_col_type_map = {c.name: c.dataType for c in df.schema}
+    for raw_column_name in ctx.raw_columns:
         raw_column = ctx.raw_columns[raw_column_name]
-        expected_types = CORTEX_TYPE_TO_ACCEPTABLE_SPARK_TYPES[raw_column["type"]]
-        actual_type = input_type_map[raw_column_name]
-        if actual_type not in expected_types:
-            logger.error("found schema:")
-            log_df_schema(df, logger.error)
-
-            raise UserException(
-                "raw column " + raw_column_name,
-                "type mismatch",
-                "expected {} but found {}".format(
-                    " or ".join(str(x) for x in expected_types), actual_type
-                ),
-            )
-        target_type = CORTEX_TYPE_TO_SPARK_TYPE[raw_column["type"]]
+        expected_cortex_type = raw_column["type"]
+        if expected_cortex_type == consts.COLUMN_TYPE_INFERRED:
+            column_type = SPARK_TYPE_TO_CORTEX_TYPE[inferred_col_type_map[raw_column_name]]
```

Comment:
Does spark read numbers as floats or doubles, and longs or ints? If not, do we need to cast int -> float and double -> long? (since we expect only float and long in our ingested dataset)

Also, is this where we end up if the user is ingesting parquet and has a column of type int (which we need to cast to float), and will that be casted?

Also, what happens if it gets inferred as a column type that we didn't enumerate (e.g. date), or if the user is ingesting a parquet which has column types that we didn't expect (also e.g. date)? In that case, should we cast to string? Or error?

Something like we have on lines 257-259 could address these. Can you also add tests to confirm these three cases?